src/lib/FreeC/IR/Subterm.hs

```diff
@@ -66,12 +68,26 @@ nullary :: b -> [a] -> Maybe b
 nullary y xs | null xs   = Just y
              | otherwise = Nothing
 
+-- | Throws an error that the subterm of the given term of the given position
+--   does not exists.
+--
+--   The error message is annotated with the given function name.
+missingPosError :: Subterm a => String -> a -> Pos -> a
+missingPosError funcName term pos =
+  error
+    $  funcName
+    ++ ": The subterm at position "
+    ++ showPretty pos
+    ++ "in term "
+    ++ showPretty term
+    ++ " does not exists."
+
 -------------------------------------------------------------------------------
 -- Direct children                                                           --
 -------------------------------------------------------------------------------
 
 -- | Type class for AST nodes with child nodes of the same type.
-class Subterm a where
+class Pretty a => Subterm a where
```

Comment:
The documentation for `Subterm` should explain why it is a subclass of `Pretty`.